config-gen/README.md

```diff
@@ -0,0 +1,81 @@
+# Config-gen
```

Comment:
I understand that the idea here is to add an alpha feature that will allow KB gen the project based on yaml file configuration instead of using the commands. 

Then, I think that needs to be a plugin. How we will know that the project was generated using this feature? WDYT about it be a plugin as well and respect the config(Project) file too? 

Could the config/gen use the https://github.com/kubernetes-sigs/kubebuilder/tree/master/pkg/plugins/golang/v3/scaffolds/internal/templates instead of its own templates? In this way, we can ensure that it is centralized and when user gen the project with this feature or by using the commands the same result would be applied.

Reply:
/hold

I am holding this one for it does not get merged before we stress what would be the best approach to add this feature. 

And then, I also think that we need to change the testdata to be by plugin:

```
testdata:
     golang/
           v2/
           v3/
     config-gen/
           v1alpha1/ 
````

Reply:
I'm not sure that we gain much by making this a plugin vs just a separate command.  It doesn't depend on any special project setup, nor does it emit any scaffolding files (it writes to stdout).

Since:

1. Existing commands write kustomize files, and this command emits config to stdout so it behaves differently
2. It doesn't change the project structure in anyway -- users just manually create their own `kubebuilderproject.yaml` file to use it

I think it makes more sense as being run as a separate command rather than an extension of an existing command.  WDYT?

Paths forward:

- Make it a subcommand of kubebuilder (as shown)
  - Could make it hidden from help
- Make it a separate executable (main.go) in the kubebuilder repo and folks `go get` it manually
- Put it in a separate repo independent of the kubebuilder project (e.g. `github.com/pwittrock/kubebuilder-fn`) and folks can `go get` it from there if they want
- Do one of the above as a plugin

Reply:
Really thank you. As we chat will

/hold cancel

I do not block this PR. And will try to check it further. However, would really nice get other thoughts as well.